### PR TITLE
Move event state from Node to Element

### PR DESCRIPTION
**Pull request by bholley**
Just getting my feet wet with Rust here. Please feel free to nit the hell out of it stylistically and idiomatically. :-)

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/7935)

### CODE REVIEW

File: components/script/dom/element.rs [hunk 1]
```diff
@@ -20,7 +20,7 @@ use dom::bindings::codegen::Bindings::HTMLTemplateElementBinding::HTMLTemplateEl
 use dom::bindings::codegen::Bindings::NamedNodeMapBinding::NamedNodeMapMethods;
 use dom::bindings::codegen::Bindings::NodeBinding::NodeMethods;
 use dom::bindings::codegen::InheritTypes::{CharacterDataCast, DocumentDerived, ElementCast};
-use dom::bindings::codegen::InheritTypes::{ElementDerived, EventTargetCast, HTMLAnchorElementCast};
+use dom::bindings::codegen::InheritTypes::{ElementDerived, EventTargetCast, HTMLAnchorElementCast, HTMLFieldSetElementDerived, HTMLLegendElementDerived, HTMLOptGroupElementDerived};
```
Comment (frewsxcv) [inline components/script/dom/element.rs]: https://travis-ci.org/servo/servo/jobs/84430060

The tidy check failed because these imports aren't sorted

Comment (frewsxcv) [inline components/script/dom/element.rs]: *All* of the imports need to be sorted, so check out the next line and notice how those fall alphabetically before some of the imports on this line. Let me know if you need help

**Comment by Ms2ger** [comment]
test-unit says:

> Your changes have increased the stack size of commonly used DOM struct Element from 304 to 312.

**Comment by nox** [comment]
@bors-servo r+

-S-awaiting-review 


---

Reviewed 14 of 14 files at r1, 2 of 2 files at r2, 4 of 4 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/7935)

**Comment by nox** [comment]
Silly me, still needs a rebase.

**Comment by nox** [comment]
I also changed how NodeFlags are initialised in #8020, this will need changes.

**Comment by bholley** [comment]
Nontrivial rebasing: r? @nox

**Comment by bholley** [comment]
-S-needs-rebase

**Comment by bholley** [comment]
@bors-servo

-S-needs-rebase

**Comment by bholley** [comment]
How do I tell bors to remove the S-needs-rebase label?

**Comment by frewsxcv** [comment]
For now, only members of the Servo github org can do that. I went ahead and did it

**Comment by bholley** [comment]
@frewsxcv What's the process for joining the Servo github org?

**Comment by frewsxcv** [comment]
Not sure what the process is, you might have better luck asking on IRC. I think I was originally added because I kept nagging people about closing old issues and removing invalid labels.

**Comment by bholley** [comment]
@pcwalton ^

**Comment by nox** [comment]
Reviewed 15 of 15 files at r4.
Review status: all files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

<sup>**[components/script/dom/element.rs, line 143 \[r4\]](https://reviewable.io:443/reviews/servo/servo/7935#-K0i3agS5KKZhnEiF9h_)** ([raw file](https://github.com/servo/servo/blob/82e10eb44e9bf76efdc9b5e50a8a37e1d2b0536c/components/script/dom/element.rs#L143)):</sup>
@Manishearth This is going to break when your lint gets fixed, right?


---

<sup>**[components/script/dom/htmlbuttonelement.rs, line 20 \[r4\]](https://reviewable.io:443/reviews/servo/servo/7935#-K0i3mWJ502tVEdlCP8f)** ([raw file](https://github.com/servo/servo/blob/82e10eb44e9bf76efdc9b5e50a8a37e1d2b0536c/components/script/dom/htmlbuttonelement.rs#L20)):</sup>
Nit: put that on the same line as `Node`.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/7935)

**Comment by Manishearth** [comment]
@nox We can allow that line or just update the lint to ignore `new_*`

**Comment by nox** [comment]
@Manishearth Silly me, I did that already in #8020.

**Comment by bholley** [comment]
Addressed nit. r? @nox

**Comment by nox** [comment]
@bors-servo r+

Thanks for your contribution.


---

Reviewed 5 of 5 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/7935)

**Comment by jdm** [comment]
```
---- focus_selector.html == focus_selector_ref.html stdout ----
	thread 'focus_selector.html == focus_selector_ref.html' panicked at 'rendering difference: /tmp/servo-reftest-000065-diff.png', 
```
That looks legit.

**Comment by bholley** [comment]
I didn't realize that Cell<T>.get() returns a copy rather than a ref. Updating.

**Comment by bholley** [comment]
Though I'm confused as to why this didn't surface before. Does CI only run when the PR Is actually about to be merged? if so, is there a tryserver-like way that I can get test results before that stage?

**Comment by jdm** [comment]
You can get a reviewer to run `@bors-servo try`.

**Comment by nox** [comment]
@bors-servo r+

Thanks for your contribution.


---

Reviewed 1 of 1 files at r6.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/7935)

**Pull request merged**